adam/learner/perception_graph_template.py

```diff
@@ -14,19 +22,12 @@
     PerceptionGraphPattern,
     raise_graph_exception,
     NodePredicate,
+    TemporalScope,
 )
 from adam.semantics import ObjectSemanticNode, SyntaxSemanticsVariable
-from attr import attrib, attrs
-from immutablecollections import (
-    ImmutableDict,
-    immutabledict,
-    immutablesetmultidict,
-    ImmutableSetMultiDict,
-)
-from immutablecollections.converter_utils import _to_immutabledict
 
 
-@attrs(frozen=True, slots=True, eq=False)
+@attrs(slots=True, eq=False)
```

Comment:
Why not frozen?

Reply:
Double checking. Likely changed during a potential bug fix